controllers/admin/AdminModulesController.php

```diff
@@ -1415,7 +1415,13 @@ public function initModal()
         $translateLinks = [];
 
         if (Tools::getIsset('configure')) {
+            /** @var Module|false $module */
             $module = Module::getInstanceByName(Tools::getValue('configure'));
+
+            if (false === $module) {
```

Comment:
@Matt75  Sorry but, this case needs to be handled more cleanly
This should throw an Exception if the module is not valid, not silently return.
I really don't understand why you ignored this comment https://github.com/PrestaShop/PrestaShop/pull/19448#discussion_r432150973 by @arouiadib 🤔 ?

Reply:
@PululuK I might side with @Matt75 on this one 😄 

I think we should fail silently here because modules not being found can happen in multiple real usecases. For example people who want to uninstall a module will sometimes `rm -rf` the folder without triggering the uninstall 😅 .

So even though in theory an exception would be better, I think it's safe here to choose code robustness against code integrity.

The idea is "even if some modules are not uninstalled properly, the merchant should be able to see his page because he needs it to manage other modules".

If we wanted to do it a lot cleaner, what we could do:
- move this logic outside the controller because parsing modules should not happen in a controller, for example in a class `ModuleParser.php`
- in `ModuleParser.php` we would raise meaningful exceptions for such usecases
- in controller, we decide how to deal with such exceptions with a `try/catch` and here we would collect those exceptions, display a warning message about it BUT still allow the page to render

Reply:
(also, @Matt75 provided a quick fix for this critical issue ; so we might merge the quick fix to "bring back" the page ; but also work on a better fix later)

Reply:
Precisely, as this is a critical case it will have to be managed more properly.
What will happen if $module is not false and is not a instance of Module class ?

Reply:
Critical bugs should be considered with serenity !

Reply:
It has been defined as critical by the author, not by the team :sweat_smile: 
The check about defined if it's an instance of Module class or not must be in `getInstanceByName`.
I agree with @matks here (and he knows it doesn't happen often :trollface: )

Reply:
SCREENSHOT

Reply:
I am a bit late for this, but just want to note that @matt75 didn't ignore my comment, it is me that  approved PR before commenting (Maybe github considers such comments as stale). Thanks @PululuK, and all of you guys, for the usual informative PRs and discussions :).

Reply:
> Precisely, as this is a critical case it will have to be managed more properly.

> Critical bugs should be considered with serenity !

Wise words indeed 👍 with the benefit of hindsight I acted too fast. I was blinded by the CRITICAL label and I wanted to fix it quickly

Thanks for the warning @PululuK 😉

Reply:
Don't worry @matks  !